### DISCUSSION TITLE
F-Droid can't build

### DISCUSSION THREAD

**Discussion by licaon-kter** [yogeshpaliyal/KeyPass#576] [General]
ref: https://gitlab.com/fdroid/fdroiddata/-/jobs/3839112546#L910

/LE: https://gitlab.com/fdroid/fdroiddata/-/commit/5353695af509eeaddd130fc5940e917c701c8f2d

**Comment by yogeshpaliyal** [comment]
Now it will build properly

**Comment by yogeshpaliyal** [comment]
@licaon-kter Do i need to do any thing to start build process?

**Comment by licaon-kter** [comment]
Depends on what "now" means. Latest commit? Latest release? 

If it's a commit or you've retagged the disabled version without bumping versionCode then something has to be updated F-Droid side.

If a new release then no, checkupdate will pick it up soon™

**Comment by yogeshpaliyal** [comment]
I ve deleted the older version (v1.4.9) and create new version (v1.4.9)

**Comment by yogeshpaliyal** [comment]
What needs to be changed on F-Droid?

**Comment by licaon-kter** [comment]
@yogeshpaliyal that disable line needs to be removed and the commit needs to be updated to the proper one

**Comment by licaon-kter** [comment]
I'm building locally to test now.

**Comment by yogeshpaliyal** [comment]
Cool thanks @licaon-kter

**Comment by licaon-kter** [comment]
done https://gitlab.com/fdroid/fdroiddata/-/commit/e35da1c1984cbaf539ee14ab77fa24a10d2b5d5c

**Comment by yogeshpaliyal** [comment]
Thanks @licaon-kter

**Comment by yogeshpaliyal** [comment]
@licaon-kter new versions are not published to F-Droid can you please help, where can I find the issue

**Comment by licaon-kter** [comment]
@yogeshpaliyal not sure, some info here: https://gitlab.com/fdroid/fdroidserver/-/issues/1115#note_1381384818

Can you remove the `latest-master` tag so I can test?

**Comment by yogeshpaliyal** [comment]
@licaon-kter removed the latest master tag

**Comment by yogeshpaliyal** [comment]
@licaon-kter can't we pick the latest release version instead of pre-release version?

**Comment by licaon-kter** [comment]
Added 1.4.12 manually for now: https://gitlab.com/fdroid/fdroiddata/-/commit/2f3a5189c83eba048e332f92558d109bee019a69

> can't we pick the latest release version instead of pre-release version?

We need a way to differentiate them, eg. keep stable as `1.4.13` but unstable as `1.4.13-alpha` and then we can use regex to pick only tags that end in numbers ;)

**Comment by yogeshpaliyal** [comment]
Got it thanks

**Comment by yogeshpaliyal** [comment]
Hey @licaon-kter can you please check why the 1.4.13 is not being published, I removed the latest-master tag

**Comment by licaon-kter** [comment]
Updated and fixed the check: https://gitlab.com/fdroid/fdroiddata/-/commit/cc4f869bdbb33714d9df0811fd992b10c478860d

**Comment by licaon-kter** [comment]
Umm, is this https://github.com/yogeshpaliyal/KeyPass/blob/v1.4.13/settings.gradle.kts#L6 really needed?

I've removed it and it builds: https://gitlab.com/fdroid/fdroiddata/-/commit/712b11b0b90589bdf958470f387aade33787dfbb

ref: https://gitlab.com/fdroid/fdroiddata/-/jobs/4300043877#L271

**Comment by yogeshpaliyal** [comment]
Why do we have to remove this? This is the open source project, and in future multiple projects can come with this.

**Comment by licaon-kter** [comment]
It's not on the list of trusted repos: https://gitlab.com/fdroid/fdroidserver/-/blob/2488cb57102582fdc3575f1c2f21705c32c85256/fdroidserver/scanner.py#L447-L467

related: https://f-droid.org/2022/07/22/maven-central.html

/LE: where can I read the policy of that repo exactly? Who pushes what and how?

**Comment by yogeshpaliyal** [comment]
Here is the one repo that publish to this repository
https://github.com/JetBrains/compose-multiplatform
These are published by Jetbrains.
https://www.jetbrains.com/community/opensource/#support

**Comment by licaon-kter** [comment]
Anyway since the libs are on Maven Central too...

>  in future multiple projects can come with this.

We'll see that when we get there :shrug: 

> Here is the one repo that publish to this repository
> These are published by Jetbrains.

Not what I've asked :slightly_smiling_face:

**Comment by yogeshpaliyal** [comment]
Can't we add check tags

**Comment by licaon-kter** [comment]
What are those?

**Comment by yogeshpaliyal** [comment]
Hi @licaon-kter can we add an update check on tags starting like these v*.*.* Only not for latest-master tag.
Or can we add a check on Releases instead of Tag, and in releases we have release and pre-release so check only ln release.

**Comment by licaon-kter** [comment]
We can't differenciate Release (of Github) vs general Tags (of git).

But... we can regex tags, eg. pickup only those that end in numbers like `1.2.3` but skip others like `1.2.4-alpha`

**Comment by yogeshpaliyal** [comment]
can you please add regex for tags eg: v1.2.3

**Comment by licaon-kter** [comment]
Done https://gitlab.com/fdroid/fdroiddata/-/commit/c8c2a3151cb7d1e4acc8414838179faf02b4199a

**Comment by yogeshpaliyal** [comment]
Thanks 🙌

**Comment by yogeshpaliyal** [comment]
@licaon-kter is there any way to rollback my android app from v1.4.18 to 1.4.17

**Comment by yogeshpaliyal** [comment]
Seeing some bugs on the latest build

**Reply by licaon-kter** [reply]
@yogeshpaliyal we can disable it, but it takes a cycle...

**Reply by yogeshpaliyal** [reply]
Please disable it for now

**Reply by licaon-kter** [reply]
Done https://gitlab.com/fdroid/fdroiddata/-/commit/19823e2b2913f18d964f0ff0f4459b8dfe145cb9

**Comment by yogeshpaliyal** [comment]
@licaon-kter can you please change build command from `assembleProductionRelease` to `assembleRelease` also the output directory as well

**Reply by licaon-kter** [reply]
@yogeshpaliyal yup https://gitlab.com/fdroid/fdroiddata/-/commit/aeee1e7891f8a7d776497a27c98ef31552523fa0

**Reply by yogeshpaliyal** [reply]
Thanks @licaon-kter

**Reply by yogeshpaliyal** [reply]
@licaon-kter Sorry for the changes again can you please change the build command from `assembleRelease ` to `assembleFreeRelease` also the output directory.

**Reply by licaon-kter** [reply]
Hmm, I didn't get a chance to change it and it was built the same :( https://monitor.f-droid.org/builds/log/com.yogeshpaliyal.keypass/1426#site-footer

will there be any issues?

**Reply by licaon-kter** [reply]
ah. now I see it, it's not in 1.4.26 yet, but in a future version?

**Reply by yogeshpaliyal** [reply]
Yes it will be in future version
